api/admin/controller.py

```diff
@@ -1614,11 +1616,106 @@ def metadata_services(self):
         if isinstance(result, ProblemDetail):
             return result
 
+        # Register this site with the Metadata Wrangler.
+        if ((is_new or not service.password) and
+            service.protocol == ExternalIntegration.METADATA_WRANGLER):
+
+            problem_detail = self.sitewide_registration(
+                service, do_get=do_get, do_post=do_post, key=key
+            )
+            if problem_detail:
+                return problem_detail
+
         if is_new:
             return Response(unicode(_("Success")), 201)
         else:
             return Response(unicode(_("Success")), 200)
 
+    def sitewide_registration(self, integration, do_get=HTTP.get_with_timeout,
+        do_post=HTTP.post_with_timeout, key=None
+    ):
+        """Performs a sitewide registration for a particular service, currently
+        only the Metadata Wrangler.
+
+        :return: A ProblemDetail or, if successful, None
+        """
+        if not integration:
```

Comment:
This method is largely the same as it was below. It takes an integration object now, which it will update with a shared secret if the exchange is successful. In the case of success, it returns `None` -- otherwise a `ProblemDetail` is passed back to be returned to the admin interface.